lib/index.js

```diff
@@ -2850,12 +2858,16 @@ addToplevelMethod('pipeline', function (/*through...*/) {
         });
     });
 
+    wrapper.writable = true;
+
     wrapper.write = function (x) {
         return start.write(x);
     };
 
     wrapper.end = function () {
-        return start.end();
+        var result = start.end();
+        wrapper._onEnd();
```

Comment:
No, I meant add `wrapper.writable = false` to [this method](https://github.com/rhendric/highland/blob/patch-1/lib/index.js#L1022). This is unrelated to `pipeline`.